otherlibs/unix/dup_win32.c

```diff
@@ -109,7 +109,8 @@ CAMLprim value caml_unix_dup2(value cloexec, value fd1, value fd2)
   }
 
   /* Reflect the dup2 on the CRT fds, if any */
-  if (CRT_fd_val(fd1) != NO_CRT_FD || CRT_fd_val(fd2) != NO_CRT_FD)
+  if (caml_win32_get_CRT_fd(fd1) != NO_CRT_FD ||
+      caml_win32_get_CRT_fd(fd2) != NO_CRT_FD)
```

Comment:
```suggestion
  if (atomic_load(&CRT_fd_val(fd1)) > NO_CRT_FD ||
      atomic_load(&CRT_fd_val(fd2)) > NO_CRT_FD)
```

Reply:
Here too, I'm not comfortable with taking a decision while one or both of fd1 and fd2 may be getting a CRT fd.